### PR TITLE
Loop around options using: next or previous

### PR DESCRIPTION
**Pull request by dhoko**
Hi,

A small fix if you select the previous/next option. 
## Before

before
## After

after

Case 1:
Auto select the first item of the list when you press the DOWN_KEY (_at the end of the list_).

Case 2:
Auto select the last item of the list when you press the UP_KEY (_at the top of the list_).

### CODE REVIEW

**Comment by LeaVerou** [comment]
Thanks for contributing!

This behavior is intentional, as an escape mechanism. But if others want this to be merged, fine by me. No strong opinions here :)

**Comment by dhoko** [comment]
Thx :) 
IMHO, it feels weird to lost the focus.
If we use a datalist, it's the same behaviour as this PR. cf:

output

File: awesomplete.js [hunk 1]
```diff
@@ -169,14 +169,15 @@ _.prototype = {
 
 	next: function () {
 		var count = this.ul.children.length;
-
-		this.goto(this.index < count - 1? this.index + 1 : -1);
+		var defaultPos = count ? 0 : -1;
+		this.goto(this.index < count - 1? this.index + 1 : defaultPos);
```
Comment (LeaVerou) [inline awesomplete.js]: No need for a variable here, please put it inline?

Comment (dhoko) [inline awesomplete.js]: I can :) 
_imho_ I don't think it's a good idea, because a ternary inside another one is not easy to read.

Comment (LeaVerou) [inline awesomplete.js]: Not even with parens? :)

Comment (dhoko) [reply awesomplete.js]: Done (_but imho still not as readable as before_).

**Comment by dhoko** [comment]
No changes with the .min after a new build :)

**Comment by LeaVerou** [comment]
Ok, merged! Thanks!

**Comment by dhoko** [comment]
:ok_hand: 

Are you going to deploy a new version (_patch ?_) via npm/bower ?

**Comment by LeaVerou** [comment]
Not just for this, but yes once there are enough changes that warrant a new version :)

**Comment by ysndmr** [comment]
Although I have made all the updates, it still doesn't open. I couldn't fix this problem.  is there someone who can help?

**Pull request merged**